lib/bindings/python/rust/llm/block_manager/vllm/connector/leader/slot.rs

```diff
@@ -630,10 +632,18 @@ impl Slot for VllmConnectorSlot {
             self.device_blocks.extend(block_ids);
         }
 
+        // This approach is fragile, but it’s the only way I can think of to skip evaluating
+        // the G1-matched blocks and to avoid offloading them again.
+        // TODO: Consider adding an indicator in the scheduler output to distinguish between
+        // matched and unmatched device blocks/tokens from the scheduler.
+        if is_new_request && computed_position > 0 && self.evaluated_blocks == 0 {
+            self.evaluated_blocks += (computed_position + 1) / self.block_size;
```

Comment:
Does this still give the right result if `computed_position` is `self.block_size - 1`?